### PR TITLE
Added API functions.

### PR DESCRIPTION
**Pull request by emilwestergren**
@sebastianbaginski @simonmika peer review

### CODE REVIEW

File: source/concurrent/Promise.ooc [hunk 1]
```diff
@@ -23,6 +23,9 @@ Promise: abstract class {
 		collector add(other)
 		collector
 	}
+	kean_concurrent_Promise_wait: unmangled func { this wait() }
+	kean_concurrent_Promise_wait_timeout: unmangled func(timeout: Double) { this wait(timeout) }
+	kean_concurrent_Promise_free: unmangled func { this free() }
```
Comment (sebastianbaginski) [inline source/concurrent/Promise.ooc]: I think we use lower case for class name in other unmangled methods.

Comment (emilwestergren) [inline source/concurrent/Promise.ooc]: Correct.

File: source/concurrent/Promise.ooc [hunk 2]
```diff
@@ -23,6 +23,9 @@ Promise: abstract class {
 		collector add(other)
 		collector
 	}
+	kean_concurrent_promise_wait: unmangled func { this wait() }
+	kean_concurrent_promise_wait_timeout: unmangled func(timeout: Double) { this wait(timeout) }
```
Comment (sebastianbaginski) [inline source/concurrent/Promise.ooc]: magic

**Comment by emilwestergren** [comment]
Updated.

**Comment by simonmika** [comment]
Go ahead, we should soon change the class name to pascal casing.

**Pull request merged**